Wikipedia/Code/HistoryViewController.swift

```diff
@@ -1,306 +1,377 @@
+        do {
+            try dataStore.viewContext.clearReadHistory()
+            viewModel.sections.removeAll()
+
+        } catch let error {
+            showError(error)
+        }
+
+        Task {
+            do {
+                let dataController = try WMFPageViewsDataController()
+                try await dataController.deleteAllPageViewsAndCategories()
+
+            } catch {
+                DDLogError("Failure deleting WMFData WMFPageViews: \(error)")
+            }
+        }
     }
 
-    override func collectionView(_ collectionView: UICollectionView, didSelectItemAt indexPath: IndexPath) {
-        guard let navigationController,
-              let articleURL = articleURL(at: indexPath) else {
-            collectionView.deselectItem(at: indexPath, animated: true)
+    private func updateProfileButton() {
+
+        guard let dataStore else {
             return
         }
-        
-        let articleCoordinator = ArticleCoordinator(navigationController: navigationController, articleURL: articleURL, dataStore: dataStore, theme: theme, source: .history)
-        articleCoordinator.start()
+
+        let config = self.profileButtonConfig(target: self, action: #selector(userDidTapProfile), dataStore: dataStore, yirDataController: yirDataController, leadingBarButtonItem: nil)
+        updateNavigationBarProfileButton(needsBadge: config.needsBadge, needsBadgeLabel: CommonStrings.profileButtonBadgeTitle, noBadgeLabel: CommonStrings.profileButtonTitle)
     }
 
-    func updateVisibleHeaders() {
-        for indexPath in collectionView.indexPathsForVisibleSupplementaryElements(ofKind: UICollectionView.elementKindSectionHeader) {
-            guard let headerView = collectionView.supplementaryView(forElementKind: UICollectionView.elementKindSectionHeader, at: indexPath) as? CollectionViewHeader else {
-                continue
-            }
-            headerView.title = titleForHeaderInSection(indexPath.section)
+    // MARK: - Reading lists hint controller
+
+    private func setupReadingListsHelpers() {
+        guard let dataStore else { return }
+        hintController = ReadingListHintController(dataStore: dataStore)
+        hintController?.apply(theme: theme)
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(userDidSaveOrUnsaveArticle(_:)),
+                                               name: WMFReadingListsController.userDidSaveOrUnsaveArticleNotification,
+                                               object: nil)
+    }
+
+    @objc private func userDidSaveOrUnsaveArticle(_ notification: Notification) {
+        guard let article = notification.object as? WMFArticle else {
+            return
         }
+
+        showReadingListHint(for: article)
     }
-    
-    override var eventLoggingCategory: EventCategoryMEP {
-        return .history
+
+    private func showReadingListHint(for article: WMFArticle) {
```

Comment:
(Optional) A change I noticed is that the reading list hint no longer hides when you scroll, so it feels stuck there until I navigate away from the screen and then return. We were probably listening for some UIScrollView events to hide it before. It would be great if we can find a way to hide upon scroll here.

Reply:
I tried intercepting the scrollView and adding a gesture recognizer, but neither worked well. I made the hint self dismiss faster (default time was 13 seconds, now it's 7). It's not ideal, but it looks less stuck.